lib/network_ssl/network_ssl_compat.c

```diff
@@ -4,10 +4,81 @@
 
 #include <openssl/ssl.h>
 
+#ifndef OPENSSL_VERSION_NUMBER
+#error "OPENSSL_VERSION_NUMBER must be defined"
+#endif
+
+/*
+ * LibreSSL claims to be OpenSSL 2.0, but (currently) has APIs compatible with
+ * OpenSSL 1.0.1g.
+ */
+#ifdef LIBRESSL_VERSION_NUMBER
+#undef OPENSSL_VERSION_NUMBER
+#define OPENSSL_VERSION_NUMBER 0x1000107fL
+#endif
+
 /* Forward declarations. */
 int ssl_compat_write_ex(SSL *, const void *, size_t, size_t *);
 int ssl_compat_read_ex(SSL *, void *, size_t, size_t *);
 
+/**
+ * ssl_compat_TLS_client_method():
+ * Create a SSL_METHOD for TLS 1.2.
+ *
+ * COMPATIBILITY: provides TLS_client_method(), but only for TLS 1.2 (not as
+ * negotiated).
+ */
+const SSL_METHOD *
+ssl_compat_TLS_client_method()
+{
+
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+	return TLSv1_2_client_method();
```

Comment:
This function is deprecated in OpenSSL 1.1.0, so in order to avoid clang warnings, I had to drag in all the `OPENSSL_VERSION_NUMBER` stuff.